GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1250,15 +1253,17 @@ void OnRevisionReadCompleted()
             {
                 if (!firstRevisionReceived && !FilterIsApplied(inclBranchFilter: true))
                 {
-                    semaphoreCurrentCommit.Wait();
-
-                    // This has to happen on the UI thread
+                    // No revisions at all received
                     this.InvokeAsync(
                             () =>
                             {
+                                if (!firstRevisionReceived)
+                                {
+                                    semaphoreCurrentCommit.Wait(cancellationToken);
+                                }
+
                                 _gridView.LoadingCompleted();
                                 bool showArtificial = AddArtificialRevisions();
-                                _gridView.LoadingCompleted();
```

Comment:
This is a merge error, causing debug builds to fail for empty repos (the actual reason I returned to this now)
It is annoying that the dialog is no longer shown, hard to get info about this.
Known but closed: https://github.com/dotnet/runtime/issues/40512

Reply:
It's (kind of) tracked under https://github.com/dotnet/winforms/issues/3739.
But debug asserts only work under a debugger, otherwise the app'd crash.

Reply:
> But debug asserts only work under a debugger, otherwise the app'd crash.

Same for Trace.Assert from what I read.
This makes it very hard to detect these types of problems, the Windows just disappear without you see why. I have many instances open and may just assume that I failed to select a widow. I will probably start using Release builds for private use. Maybe we should remove all asserts.

Reply:
I don't think removing asserts is a good idea, they have purpose. Hopefully one day Windows Desktop can fix this.
I do generally run private builds in release though.